bluebrain/repo-patches/packages/neuron/package.py

```diff
@@ -1,488 +1,178 @@
-            "codegenopt",
-            default=False,
-            description="Use NMODL with codedgen ionvar copies optimizations",
-            when=nmodl_spec,
-        )
-        variant(
-            "sympy",
-            default=False,
-            description="Use NMODL with SymPy to solve ODEs",
-            when=nmodl_spec,
-        )
-        variant(
-            "sympyopt",
-            default=False,
-            description="Use NMODL with SymPy Optimizations",
-            when=nmodl_spec,
-        )
+    # used for debugging purposes
     variant(
         "prcellstate",
         default=False,
         description="Enable tracking of voltage and conductivity with prcellstate on CoreNEURON",
     )
 
-    # Build with `ninja` instead of `make`
-    generator = "Ninja"
-    depends_on("ninja", type="build")
-
-    depends_on("bison", type="build")
-    depends_on("caliper+mpi", type=("build", "link", "run"), when="+caliper+mpi")
-    depends_on("caliper~mpi", type=("build", "link", "run"), when="+caliper~mpi")
-    depends_on("flex@2.6:", type="build")
-
-    # Readline became incompatible with Mac so we use neuron internal readline.
-    # HOWEVER, with the internal version there is a bug which makes
-    # Vector.as_numpy() not work!
-    depends_on("readline", when=sys.platform != "darwin")
-
-    # Transient dependency
-    depends_on("gettext")
-
-    depends_on("mpi", when="+mpi")
-    depends_on("py-mpi4py", when="+mpi+python+tests")
-    depends_on("ncurses")
-    depends_on("python@2.6:", when="+python", type=("build", "link", "run"))
-    depends_on("py-pytest", when="+python+tests")
-    # Numpy is required for Vector.as_numpy()
-    depends_on("py-numpy", when="+python", type=("build", "run"))
-    depends_on("py-cython", when="+rx3d", type="build")
-    depends_on("py-pytest-cov", when="+tests@8:")
-    # TODO: newer spack doesn't propogate python package as dependency
-    # (as PYTHONPATH) during build time
-    depends_on("py-sympy@1.3:", type=("build", "run"))
-    # Next two needed in versions containing neuronsimulator/nrn#2235.
-    depends_on("py-pip", type=("build"), when="@9:")
-    depends_on("py-packaging", type=("run"), when="@9:")
-
-    # dependency on coreneuron via submodule
-    depends_on("coreneuron+legacy-unit~caliper", when="@:8.99+coreneuron+legacy-unit~caliper")
-    depends_on("coreneuron~legacy-unit~caliper", when="@:8.99+coreneuron~legacy-unit~caliper")
-    depends_on("coreneuron+legacy-unit+caliper", when="@:8.99+coreneuron+legacy-unit+caliper")
-    depends_on("coreneuron~legacy-unit+caliper", when="@:8.99+coreneuron~legacy-unit+caliper")
-    conflicts("coreneuron", when="@9:")
-
-    # dependencies from coreneuron package
-    depends_on("python", type=("build", "run"))
-    depends_on("boost", when="@8.99:+tests+coreneuron")
-    depends_on("cuda", when="@8.99:+gpu")
-    depends_on("libsonata-report", when="@8.99:+report+coreneuron")
+    # used furing development by core team
```

Comment:
thanks for spotting that! 😒 

I will see if I need to push another commit otherwise I will merge it.

A bit painful thing has been that neuron-check CI is failing "randomly" only with GPU builds. If I retrigger that on different  GPU nodes (especially non-ldu* / phase2) then everything is passing).

I am checking this by explicitly launching neuron CI on this branch: https://github.com/neuronsimulator/nrn/pull/2646.

If that will be green and if re-triggered CI is green than I will merge this.

Reply:
addressed as I need to fix something anyway.